Drivers/Src/Contactors.c

```diff
@@ -1,66 +1,72 @@
 /* Copyright (c) 2020 UT Longhorn Racing Solar */
 
-/**
- * Source file to mimic GPIO port communication
- */ 
 
 #include "Contactors.h"
+#include "stm32f4xx_gpio.h"
+
+#define CONTACTORS_PORT         PORTC
+#define ARRAY_PRECHARGE_PIN GPIO_Pin_10
+#define ARRAY_CONTACTOR_PIN GPIO_Pin_11
+#define MOTOR_CONTACTOR_PIN GPIO_Pin_12
+
 
 /**
  * @brief   Initializes contactors to be used
  *          in connection with the Motor and Array
- * @param   contactor the contactor to initialize
  * @return  None
  */ 
-void Contactors_Init(contactor_t contactor) {
-    BSP_GPIO_Init(CONTACTORS_PORT, 0x6, 1); //Pins 1 and 2 in Port C are output (0b110)
+void Contactors_Init() {
+    BSP_GPIO_Init(CONTACTORS_PORT, 
+                 (ARRAY_CONTACTOR_PIN) | 
+                 (ARRAY_PRECHARGE_PIN) |
+                 (MOTOR_CONTACTOR_PIN), 
+                  1);
 }
 
 /**
- * @brief   Reads contactor states from a file 
- *          and returns the current state of 
+ * @brief   Returns the current state of 
  *          a specified contactor
- * @param   contactor specifies the contactor for 
- *          which the user would like to know its state (MOTOR/ARRAY)
+ * @param   contactor the contactor
+ *              (MOTOR_PRECHARGE/ARRAY_PRECHARGE/ARRAY_CONTACTOR)
  * @return  The contactor's state (ON/OFF)
  */ 
 State Contactors_Get(contactor_t contactor) {
-    uint16_t data = BSP_GPIO_Read(CONTACTORS_PORT);
+    State state = OFF;
     switch (contactor) {
-        case MOTOR:
-            data = (data >> MOTOR_CNCTR_PIN) & 0x01;
+        case ARRAY_CONTACTOR :
+            state = BSP_GPIO_Read_Pin(CONTACTORS_PORT, ARRAY_CONTACTOR_PIN);
```

Comment:
I don't like setting the State type variable to a 1/0 directly, I would do something like the following:
```suggestion
             if( BSP_GPIO_Read_Pin(CONTACTORS_PORT, ARRAY_CONTACTOR_PIN)){
                   state = ON;
             } else {
                  state = OFF;
             }
```